composer/modules/web/src/plugins/import-struct/handlers.js

```diff
@@ -0,0 +1,139 @@
+}
+
+/**
+ * Check given value is an Integer
+ * @param  {Number} val value to check
+ * @return {Boolean}  is Intger
+ */
+function isFloat(val) {
+    return !isNaN(val) && !this.isInt(Number(val)) && val.toString().length > 0;
+}
+
+/**
+ * process JSON and generate struct fields
+ * @param  {object} parent      parent struct object
+ * @param  {object} literalExpr JSON expression
+ * @return {boolean}             status
+ */
+function processJSONStruct(parent, literalExpr) {
+    let currentValue;
+    let success = true;
+    parent.setFields([], true);
+    literalExpr.keyValuePairs.every((ketValPair) => {
+        let currentName;
+        if (TreeUtils.isLiteral(ketValPair.getKey())) {
+            currentName = ketValPair.getKey().getValue().replace(/"/g, '');
+        } else {
+            currentName = ketValPair.getKey().getVariableName().getValue();
+        }
+        if (TreeUtils.isRecordLiteralExpr(ketValPair.getValue())) {
+            const parsedJson = FragmentUtils.parseFragment(
+                            FragmentUtils.createStatementFragment(`struct { } ${currentName};`));
+            const anonStruct = TreeBuilder.build(parsedJson);
+            success = this.processJSONStruct(anonStruct.getVariable().getTypeNode().anonStruct,
+                        ketValPair.getValue());
+            if (success) {
+                parent.addFields(anonStruct.getVariable());
+            }
+            return success;
+        } else if (TreeUtils.isLiteral(ketValPair.getValue())) {
+            currentValue = ketValPair.getValue().getValue();
+            let currentType = 'string';
+            if (isInt(currentValue)) {
+                currentType = 'int';
+            } else if (isFloat(currentValue)) {
+                currentType = 'float';
+            } else if (currentValue === 'true' || currentValue === 'false') {
+                currentType = 'boolean';
+            }
+            const refExpr = TreeBuilder.build(FragmentUtils.parseFragment(
+                FragmentUtils.createStatementFragment(`${currentType} ${currentName} = ${currentValue};`)));
+            if (!refExpr.error) {
+                parent.addFields(refExpr.getVariable());
+            } else {
+                success = false;
+            }
+            return success;
+        } else {
+            success = false;
+        }
+    });
+    return success;
+}
+
+/**
+ * Provides command handler definitions of debugger plugin.
+ * @param {plugin} plugin plugin instance
+ * @returns {Object[]} command handler definitions.
+ *
+ */
+export function getHandlerDefinitions(plugin) {
+    return [
+        {
+            cmdID: COMMANDS.SHOW_IMPORT_STRUCT_DIALOG,
+            handler: () => {
+                const topLevelNodes = plugin.appContext.editor.getActiveEditor().getProperty('ast');
+                let structNode = DefaultNodeFactory.createStruct();
```

Comment:
@DimalChandrasiri ; can't we make this "const" as its not initialized?

Reply:
fixed with commit 0974ece26d1cb2557d0f080e28c729a8122f138d